server/client.go

```diff
@@ -3188,8 +3273,9 @@ func (c *client) processPingTimer() {
 		// Check for violation
 		if c.ping.out+1 > c.srv.getOpts().MaxPingsOut {
 			c.Debugf("Stale Client Connection - Closing")
-			c.sendProto([]byte(fmt.Sprintf("-ERR '%s'\r\n", "Stale Connection")), true)
-			c.clearConnection(StaleConnection)
+			c.enqueueProto([]byte(fmt.Sprintf("-ERR '%s'\r\n", "Stale Connection")))
+			c.mu.Unlock()
```

Comment:
Did this dangle holding the lock forever before??

Reply:
No, just that I need to call c.closeConnection() now and so need to release the lock (used to be a lock/defer unlock).

Reply:
Still not following, I just saw the unpaired unlock so was confused. Do you re-acquire?

Reply:
Top used to be lock/defer unlock and here we used to have c.sendProto()/c.clearConnection() and return. I am now explicit with lock/unlock because I need to call closeConnection() (line below this one) and that requires to be done outside of lock.

Reply:
ok thanks for clarifications.